include/support.h

```diff
@@ -38,6 +39,14 @@
 #include <strings.h>
 #endif
 
+// Clamp: given a value that can be compared with the given minimum and maximum
+//        values, this function will:
+//          * return the value if it's in-between or equal to either bounds, or
+//          * return either bound depending on which bound the value is beyond
+template <class T> T clamp(const T& n, const T& lower, const T& upper) {
+	return (std::max)(lower, (std::min)(n, upper));
```

Comment:
wrapping the functions with parentheses allows msvc to build cleanly. add as comment.

Reply:
I suspect forcing template parameter with:

    std::max<T>(lower, std::min<T>(n, upper));

might work as well.

Reply:
Will try it out!

Reply:
This worked; great hunch! Fixed in 03d870b